### PR TITLE
passkey provider

### PR DESCRIPTION
**Pull request by andreigiura**
### Issue/Feature
Add passkey provider and replace MetamaskProxyProvider with IframeProvider

### Reproduce
Issue exists on version `2.` of sdk-dapp.

### Contains breaking changes
[] No

[x] Yes

### Updated CHANGELOG
[x] Yes

### Testing
[x] User testing
[] Unit tests

### CODE REVIEW

**Review by razvantomegea — APPROVED** [review]

File: src/UI/iframe/IframeLoginButton/IframeButton.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import React, { ReactNode } from 'react';
 import { withStyles, WithStylesImportType } from 'hocs/withStyles';
-import { useMetamaskProxyLogin } from 'hooks/login/useMetamaskProxyLogin';
+import { useIframeLogin } from 'hooks/login/useIframeLogin';
```
Comment (CiprianDraghici) [inline src/UI/iframe/IframeLoginButton/IframeButton.tsx:3]: useWebWalletIFrame

File: src/UI/iframe/IframeLoginButton/IframeButton.tsx [hunk 2]
```diff
@@ -15,8 +15,8 @@ export interface MetamaskProxyLoginButtonPropsType
   disabled?: boolean;
 }
 
-const MetamaskProxyLoginButtonComponent: (
-  props: MetamaskProxyLoginButtonPropsType & WithStylesImportType
+const IframeLoginButtonComponent: (
+  props: IframeLoginButtonPropsType & WithStylesImportType
```
Comment (CiprianDraghici) [inline src/UI/iframe/IframeLoginButton/IframeButton.tsx:19]: IWebWalletIFrameLogin...

File: src/UI/iframe/IframeLoginButton/IframeButton.tsx [hunk 3]
```diff
@@ -53,15 +53,9 @@ const MetamaskProxyLoginButtonComponent: (
...
-    ssrStyles: () =>
-      import(
-        'UI/metamaskProxy/MetamaskProxyLoginButton/metamaskProxyLoginButton.styles.scss'
-      ),
-    clientStyles: () =>
-      require('UI/metamaskProxy/MetamaskProxyLoginButton/metamaskProxyLoginButton.styles.scss')
-        .default
-  }
-);
+export const IframeButton = withStyles(IframeLoginButtonComponent, {
```
Comment (CiprianDraghici) [inline src/UI/iframe/IframeLoginButton/IframeButton.tsx:56]: WebWalletIFrameLoginButton

**Review by CiprianDraghici — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -167,7 +167,7 @@
     "@multiversx/sdk-wallet": "4.2.0",
     "@multiversx/sdk-wallet-connect-provider": "4.1.3",
     "@multiversx/sdk-web-wallet-cross-window-provider": "1.0.0",
-    "@multiversx/sdk-web-wallet-iframe-provider": "1.0.0",
+    "@multiversx/sdk-web-wallet-iframe-provider": "^1.0.2",
```
Comment (CiprianDraghici) [inline package.json:170]: fix version

**Review by CiprianDraghici — APPROVED** [review]

**Pull request merged**